lib/cipher.rb

```diff
@@ -40,6 +40,46 @@ def self.type_s(type)
     end
   end
 
+  # shortcut to turn any field containing json data into an object
+  def method_missing(method, *args, &block)
+    if m = method.to_s.match(/^(.+)_unjson$/)
+      j = self.send(m[1])
+      j ? JSON.parse(j) : nil
+    else
+      super
+    end
+  end
+
+  # migrate from older style everything-in-data to separate fields
+  def migrate_data!
+    return if !self.data
+
+    js = JSON.parse(self.data)
+    return if !js
+
+    self.name = js.delete("Name")
+    self.notes = js.delete("Notes")
+    self.fields = js.delete("Fields").try(:to_json)
+
+    if self.type == TYPE_LOGIN
+      js["Uris"] = [
+        { "Uri" => js["Uri"], "Match" => nil },
+      ]
+      js.delete("Uri")
+    end
+
+    # move the remaining fields into the new dedicated field based on the type
+    fmap = {
+      TYPE_LOGIN => "login",
+      TYPE_NOTE => "securenote",
+      TYPE_CARD => "card",
+      TYPE_IDENTITY => "identity",
+    }
+    self.send("#{fmap[self.type]}=", js.to_json)
+
+    self.save || raise("failed migrating #{self.inspect}")
+  end
+
```

Comment:
Wouldn't this make more sense to be done in the migration, since this is not needed in normal operation: Add the method in the migration and only if needed, this shouldn't be a problem thanks to open classes.

Reply:
I don't think the database driver is the right place for it either, it should really be in a separate migration file.

At some point I'm going to stop trying to re-invent ActiveRecord and just swap that in for `db.rb` and `dbmodel.rb` and bring in [standalone_migrations](https://github.com/thuss/standalone-migrations).

Reply:
I was actually only inches away from bringing in AR in my additions, but didn't want to add it as an external dependency, since it looked like you'd prefer to have as few as possible external deps. :) 

Gonna update my additions branch with these changes and then might give it a try with bringing in active-record.